testng-core/src/main/java/org/testng/TestTaskExecutor.java

```diff
@@ -51,31 +48,21 @@ public void execute() {
     String name = "test-" + xmlTest.getName();
     int threadCount = Math.max(xmlTest.getThreadCount(), 1);
     if (RuntimeBehavior.favourCustomThreadPoolExecutor()) {
```

Comment:
Same here. Clean the code by removing the now useless runtime behavior

Reply:
@juherr - Are you saying that we can get rid of this entire if condition across the board (both at test level and suite level wherein we error out if the user tries to use the earlier deprecated GraphThreadPoolExecutor) ?

Reply:
Yes because it is 2 different logics with different responsability